CMakeLists.txt

```diff
@@ -4,77 +4,26 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 project(gbs LANGUAGES CXX)
 
-#
 # c++ conf
-#
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
-#
-# set output dirs (vs is a pain)
-#
-function(SET_OUT_DIR REQUIRED_ARG)
-    list(GET ARGV 0 TARGET_NAME)
-    list(GET ARGV 1 OUT_DIR)
-    message(${TARGET_NAME})
-    message(${OUT_DIR})
-    foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
-        string( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )
-        set_property(TARGET ${TARGET_NAME} PROPERTY RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${OUT_DIR} )
-        set_property(TARGET ${TARGET_NAME} PROPERTY LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${OUT_DIR} )
-        set_property(TARGET ${TARGET_NAME} PROPERTY ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${OUT_DIR} )
-    endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
-endfunction()
-#
-# CODAN ENV
-#
-if(EXISTS $ENV{LIBRARY_PREFIX})
-    file(TO_CMAKE_PATH $ENV{LIBRARY_BIN} LIBRARY_BIN)
-    file(TO_CMAKE_PATH $ENV{LIBRARY_INC} LIBRARY_INC)
-    file(TO_CMAKE_PATH $ENV{LIBRARY_LIB} LIBRARY_LIB)
-elseif(EXISTS $ENV{CONDA_PREFIX})
-    file(TO_CMAKE_PATH $ENV{CONDA_PREFIX} CONDA_PREFIX)
-    file(TO_CMAKE_PATH ${CONDA_PREFIX}/Library/bin LIBRARY_BIN)
-    file(TO_CMAKE_PATH ${CONDA_PREFIX}/Library/include LIBRARY_INC)
-    file(TO_CMAKE_PATH ${CONDA_PREFIX}/Library/lib LIBRARY_LIB)
-else()
-    message(FATAL_ERROR "A conda environement is required")
-endif()
-
-message(${LIBRARY_LIB} ${LIBRARY_BIN} ${LIBRARY_INC})
-#
-# set install dir
-#
-function(SET_INSTALL_DIR REQUIRED_ARG)
-    list(GET ARGV 0 TARGET_NAME)
-    if(WIN32)
-        if(TARGET ${TARGET_NAME} )
-            install(TARGETS ${TARGET_NAME}
-                EXPORT  ${TARGET_NAME}Targets
-                COMPONENT python
-                LIBRARY  DESTINATION ${LIBRARY_LIB}
-                RUNTIME  DESTINATION ${LIBRARY_BIN}
-                INCLUDES DESTINATION ${LIBRARY_INC}
-                ARCHIVE  DESTINATION ${LIBRARY_LIB}
-            )
-            install(
-                EXPORT  ${TARGET_NAME}Targets
-                FILE ${TARGET_NAME}.cmake
-                NAMESPACE GBS::
-                DESTINATION ${LIBRARY_LIB}/cmake/${TARGET_NAME}
-            )
-        endif(TARGET ${TARGET_NAME} )
-    else() # TODO find export dir for linux
-            install(TARGETS ${TARGET_NAME}  
-            LIBRARY DESTINATION $ENV{LD_RUN_PATH}
-            )
-    endif()
+
+# Set installation directories (CMAKE_INSTALL_INCLUDEDIR, CMAKE_INSTALL_BINDIR, CMAKE_INSTALL_LIBDIR)
+include(GNUInstallDirs)
+
+message("-- CMake install prefix: " ${CMAKE_INSTALL_PREFIX})
+message("  -> binaries: " ${CMAKE_INSTALL_BINDIR})
+message("  -> libs: " ${CMAKE_INSTALL_LIBDIR})
+message("  -> includes: " ${CMAKE_INSTALL_INCLUDEDIR})
+
+function(INSTALL_HEADERS REQUIRED_ARG)
```

Comment:
TARGET_NAME argument is missing
```cmake
function(INSTALL_HEADERS REQUIRED_ARG)
    # copy headers keeping file structure
    list(GET ARGV 0 TARGET_NAME)
    string(TOLOWER ${TARGET_NAME} inc_sub_dir)
    install(
            DIRECTORY "${CMAKE_SOURCE_DIR}/${inc_sub_dir}"
            DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
            FILES_MATCHING PATTERN "*.h"
            PATTERN "cmake" EXCLUDE
            PATTERN "tests" EXCLUDE
            PATTERN "build*" EXCLUDE
            PATTERN "python" EXCLUDE
            PATTERN "xml" EXCLUDE
            PATTERN "doc" EXCLUDE
            PATTERN "recipe" EXCLUDE
            PATTERN ".*" EXCLUDE
    )
endfunction()
```

Reply:
TARGET_NAME is not used anymore

Reply:
Yest but it was a trick to use name spaces that way:  target_link(TARGET_NAME GBS::MODULE_NAME)

BTW you can use this function that way but sub project have to be renamed that way gbs-render -> RENDER
and change TARGET OUTPUT_NAME.

Moreover it would more clear if INSTALL_HEADERS would be called without argument since you don't use it.

Reply:
My mistake, that way the include dir will be upercase, not good, this is why I prefer the code I proposed more over it works with the proposal I made for install